client/components/signup-form/index.jsx

```diff
@@ -311,13 +311,13 @@ export default React.createClass( {
 						autoCapitalize="off"
 						autoCorrect="off"
 						className="signup-form__input"
-						disabled={ this.state.submitting || this.props.disabled }
+						disabled={ this.state.submitting || this.props.disabled || ! this.props.allowEmailChange }
 						id="email"
 						name="email"
 						type="email"
 						value={ formState.getFieldValue( this.state.form, 'email' ) }
 						isError={ formState.isFieldInvalid( this.state.form, 'email' ) }
```

Comment:
Do we need to add the `this.props.allowEmailChange` here in the case that there is an error state?